src/main/java/com/google/sps/servlets/MemesServlet.java

```diff
@@ -26,12 +26,46 @@
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+    Part filePart = request.getPart("image");
+    String fileName = UUID.randomUUID().toString();
+    InputStream fileInputStream = filePart.getInputStream();
 
+    String uploadedFileUrl = uploadToCloudStorage(fileName, fileInputStream);
+
+    PrintWriter out = response.getWriter();
+    out.println("<p>Your Meme:</p>");
+    out.println("<a href=\"" + uploadedFileUrl + "\">");
+    out.println("<img src=\"" + uploadedFileUrl + "\" />");
+    out.println("</a><br>");
```

Comment:
Great idea to create the variable "out" referencing "response.getWriter()". It cleans up the code a lot

Reply:
+1